csrc/host_ir/executor.cpp

```diff
@@ -79,6 +79,11 @@ std::vector<at::Tensor> HostIrExecutor::runWithInput(
 
 void HostIrExecutor::handle(SetCurrentStream* set_current_stream) {
   Stream* stream = set_current_stream->stream();
+  if (Val* index = stream->index(); index != nullptr) {
+    auto value = expr_evaluator_.evaluate(index);
+    NVF_ERROR(value.hasValue() && value.is<int64_t>());
+    stream = (Stream*)(value.as<int64_t>());
```

Comment:
This is not very clean but that's the simplest way I can think of to achieve using either a `Stream*` or an `int64_t` as a key to the hashtable `streams_`. Maybe using `std::variants` or `dynamic_type` with a custom hash would be preferable, but more cumbersome, so lmk what you think is best